scripts/release.sh

```diff
@@ -0,0 +1,141 @@
+#!/bin/sh
+
+set -e
+
+LAST_TAG=`git describe --tags --abbrev=0`
+COMMITS=`git log $LAST_TAG..HEAD --pretty=oneline --abbrev-commit`
+
+if [ "$COMMITS" = "" ]; then
+  echo "Release aborted: I see no work."
+  exit 1
+fi
+
+case "$1" in
+  "")
+    BUILD_TYPE="Train"
+    ;;
+  "patch")
+    BUILD_TYPE="Patch"
+    ;;
+  *)
+    echo "Release aborted: Invalid argument \"$1\"."
+    exit 1
+    ;;
+esac
+
+while read -r COMMIT; do
+  HASH=`echo "$COMMIT" | cut -d ' ' -f 1`
+  MESSAGE=`echo "$COMMIT" | cut -d ':' -f 2- | awk '{$1=$1};1'`
+  TYPE=`echo "$COMMIT" | cut -d ' ' -f 2 | awk '{$1=$1};1' | cut -d ':' -f 1 | cut -d '(' -f 1 | awk '{$1=$1};1'`
+  AREA=`echo "$COMMIT" | cut -d '(' -f 2 | cut -d ')' -f 1 | awk '{$1=$1};1'`
+
+  if [ "$AREA" = "$COMMIT" ]; then
+    AREA=""
+  fi
+
+  if [ "$AREA" != "" ]; then
+    AREA="$AREA: "
+  fi
+
+  case "$TYPE" in
+    "feat")
+      if [ "$FEAT_SUMMARY" = "" ]; then
+        FEAT_SUMMARY="### New features\n"
+      fi
+      FEAT_SUMMARY="$FEAT_SUMMARY\n* $AREA$MESSAGE ($HASH)"
+      ;;
+    "fix")
+      if [ "$FIX_SUMMARY" = "" ]; then
+        FIX_SUMMARY="### Bug fixes\n"
+      fi
+      FIX_SUMMARY="$FIX_SUMMARY\n* $AREA$MESSAGE ($HASH)"
+      ;;
+    "perf")
+      if [ "$PERF_SUMMARY" = "" ]; then
+        PERF_SUMMARY="### Performance improvements\n"
+      fi
+      PERF_SUMMARY="$PERF_SUMMARY\n* $AREA$MESSAGE ($HASH)"
+      ;;
+    "refactor")
+      if [ "$REFACTOR_SUMMARY" = "" ]; then
+        REFACTOR_SUMMARY="### Refactorings\n"
+      fi
+      REFACTOR_SUMMARY="$REFACTOR_SUMMARY\n* $AREA$MESSAGE ($HASH)"
+      ;;
+    "revert")
+      if [ "$REFACTOR_SUMMARY" = "" ]; then
+        REVERT_SUMMARY="### Reverted changes\n"
+      fi
+      REVERT_SUMMARY="$REVERT_SUMMARY\n* $AREA$MESSAGE ($HASH)"
+      ;;
+    *)
+      if [ "$OTHER_SUMMARY" = "" ]; then
```

Comment:
We are probably inconsistent with it but other repos have a `chore` section. Not sure if worthwhile to have it in its own for put in other section.

Reply:
I figured `chore` makes sense to go under "Other changes", they'll mostly be `chore` items but occasionally other stuff might pop up in there too.